crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -443,26 +428,49 @@ pub fn derive_system_param(input: TokenStream) -> TokenStream {
 
     TokenStream::from(quote! {
         pub struct #fetch_struct_name<#punctuated_generics>(#phantoms);
-        impl #impl_generics #path::SystemParam for #struct_name#ty_generics #where_clause {
+        impl #impl_generics #crate_path::SystemParam for #struct_name#ty_generics #where_clause {
             type Fetch = #fetch_struct_name <#punctuated_generic_idents>;
         }
 
-        impl #impl_generics #path::FetchSystemParam<'a> for #fetch_struct_name<#punctuated_generic_idents> {
+        impl #impl_generics #crate_path::FetchSystemParam<'a> for #fetch_struct_name<#punctuated_generic_idents> {
             type Item = #struct_name#ty_generics;
-            fn init(system_state: &mut #path::SystemState, world: &#path::World, resources: &mut #path::Resources) {
-                #(<<#field_types as SystemParam>::Fetch as #path::FetchSystemParam>::init(system_state, world, resources);)*
+            fn init(system_state: &mut #crate_path::SystemState, world: &#crate_path::World, resources: &mut #crate_path::Resources) {
+                #(<<#field_types as SystemParam>::Fetch as #crate_path::FetchSystemParam>::init(system_state, world, resources);)*
             }
 
             unsafe fn get_param(
-                system_state: &'a #path::SystemState,
-                world: &'a #path::World,
-                resources: &'a #path::Resources,
+                system_state: &'a #crate_path::SystemState,
+                world: &'a #crate_path::World,
+                resources: &'a #crate_path::Resources,
             ) -> Option<Self::Item> {
                 Some(#struct_name {
-                    #(#fields: <<#field_types as SystemParam>::Fetch as #path::FetchSystemParam>::get_param(system_state, world, resources)?,)*
+                    #(#fields: <<#field_types as SystemParam>::Fetch as #crate_path::FetchSystemParam>::get_param(system_state, world, resources)?,)*
                     #(#ignored_fields: <#ignored_field_types>::default(),)*
                 })
             }
         }
     })
 }
+
+fn bevy_ecs_path() -> Path {
```

Comment:
I don't like that `Path` is overloaded here, but that's `syn`'s fault not yours

Reply:
I qualified the Path as `syn::Path`, I agree that reusing `Path` is confusing here.